metsim/metsim.py

```diff
@@ -580,8 +598,56 @@ def wrap_run(func: callable, loc: dict, params: dict,
     lat = ds['lat'].values
     elev = ds['elev'].values
     swe = ds['swe'].values
+
+    if params['prec_type'].upper() == 'TRIANGLE':
+        try:
+            dur = ds['dur'].values
+        except:
+            print('Storm duration and time to peak values are required in the '
+                  'domain file for the triangle preciptation disagregation '
+                  'method.')
+            sys.exit()
+        try:
+            t_pk = ds['t_pk'].values
+        except:
+            print('Storm duration and time to peak values are required in the '
+                  'domain file for the triangle preciptation disagregation '
+                  'method.')
+            sys.exit()
+        day_length = cnst.MIN_PER_HOUR * cnst.HOURS_PER_DAY
+        dur_zero_test = dur <= 0
+        dur_day_test = dur > day_length
+        if dur_zero_test.any() or dur_day_test.any():
+            print('Storm duration must be greater than 0 and less than',
+                  day_length, '(i.e. the day length in minutes)')
+            sys.exit()
+        t_pk_zero_test = t_pk < 0
+        t_pk_day_test = t_pk > day_length
+        if t_pk_zero_test.any() or t_pk_day_test.any():
+            print('Storm time to peak must be greater than or equal to 0, and'
+                  'less than', day_length, '(i.e. the end of a day)')
+            sys.exit()
+
     params['elev'] = elev
+
     df = ds.to_dataframe()
+
+    if params['prec_type'].upper() == 'TRIANGLE':
+        tri_df = df.copy(deep=True)
+        vars_to_drop = list(tri_df)
+        vars_to_drop.remove('t_pk')
+        vars_to_drop.remove('dur')
+        tri_df = tri_df.drop(labels=vars_to_drop, axis=1)
+        tri_df.reset_index(level='time', drop=True, inplace=True)
+        tri_df = tri_df.reset_index().drop_duplicates(subset='month',
+                                                      keep='first').set_index('month')
+        months = np.arange(12, dtype=int)
+        tri_df.index = months
+        df = df.drop(['dur', 't_pk'], axis=1)
+        df.reset_index(level='month', drop=True, inplace=True)
+        df = df.reset_index().drop_duplicates(subset='time',
+                                              keep='first').set_index('time')
```

Comment:
added:

# Drop monthly triangle variables from df so that df only contains the
# full timeseries data
df = df.drop(['dur', 't_pk'], axis=1)

# Remove the monthly subset index from df so that df does not contain
# a multi-level index and the index is just the timeseries of dates
df.reset_index(level='month', drop=True, inplace=True)
df = df.reset_index().drop_duplicates(subset='time',
                                                          keep='first').set_index('time')

Reply:
This section of the code was removed.